Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/MainAddActionsNavigationPage.xaml.cs

```diff
@@ -15,14 +24,62 @@ namespace Ginger.BusinessFlowsLibNew.AddActionMenu
     /// </summary>
     public partial class MainAddActionsNavigationPage : Page
     {
-        IWindowExplorer WindowExplorerDriver;
         Context mContext;
+        RecordNavPage mRecordPage = null;
+        SharedRepositoryNavPage mSharedRepositoryNavPage = null;
+        POMNavPage mPOMNavPage = null;
+        ActionsLibraryNavPage mActionsLibraryNavPage = null;
+        LiveSpyNavPage mLiveSpyNavPage = null;
+        WindowsExplorerNavPage mWindowsExplorerNavPage = null;
+
         public MainAddActionsNavigationPage(Context context)
         {
             mContext = context;
             InitializeComponent();
+            context.PropertyChanged += Context_PropertyChanged;
             xNavigationBarPnl.Visibility = Visibility.Collapsed;
             xSelectedItemFrame.ContentRendered += NavPnlActionFrame_ContentRendered;
+
+            ePlatformType ePlatformType = (from x in WorkSpace.Instance.Solution.ApplicationPlatforms
+                                           where x.AppName == mContext.BusinessFlow.CurrentActivity.TargetApplication
+                                           select x.Platform).FirstOrDefault();
+
+            if (PlatformInfoBase.GetPlatformImpl(ePlatformType) != null && PlatformInfoBase.GetPlatformImpl(ePlatformType).IsPlatformSupportPOM())
+            {
+                xRecordItemBtn.IsEnabled = true;
+            }
+            else
+            {
+                xRecordItemBtn.IsEnabled = false;
+            }
+        }
+
+        private void Context_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e != null && (e.PropertyName == nameof(BusinessFlow) || e.PropertyName == nameof(Activity)))
+            {
+                xRecordItemBtn.IsEnabled = false;
+                ePlatformType ePlatformType = ePlatformType.NA;
+
```

Comment:
don't take first appAgent if CurrentActivity is null- if it is null so we can't add Actions and all panel should be disabeled